File: lib/features/presentation/login/models/water_model.dart
class Water {
  String date;
  String firebaseUid;
  bool completed;
  int peeCount;

  Water({
    this.date = '',
    this.firebaseUid = '',
    this.completed = false,
    this.peeCount = 0,
  });

  factory Water.fromJson(Map<String, dynamic> json) {
    return Water(
      date: json['date'] ?? '',
      firebaseUid: json['firebase_uid'] ?? '',
      completed: json['completed'] ?? false,
      peeCount: json['pee_count'] ?? 0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'date': date,
      'firebase_uid': firebaseUid,
      'completed': completed,
      'pee_count': peeCount,
    };
  }
}

File: lib/backend/app/server/models/water_model.py
from beanie import Document
from typing import List, Optional
from beanie import PydanticObjectId

class Water(Document):
    class Settings:
        collection = "Water"
    day_id: PydanticObjectId # Reference to the Day collection
    pee_count: int
    amount_finished: int
    completed: bool




# OLD VERSION (12/19)
# from beanie import Document
# from pydantic import EmailStr
# from typing import List, Optional
# from datetime import datetime
# class Water(Document):
#     class Settings:
#         name = "Water"
#     date : str  # Date of the day
#     firebase_uid : str
#     completed: bool 
#     peeCount: int
#     completed: bool
#     ouncesDrunk: int

    
